custom_components/ha_creality_ws/utils.py

```
@@ -171,6 +171,15 @@ class ModelDetection:
     Provides capability flags and a resolved model name if possible.
     """
     
+    # Klipper output_pin name per model for LED brightness (SET_PIN dimming).
+    # Keys are ModelDetection boolean-attribute names (evaluated in order); the
+    # value is the Klipper output_pin name. Extend this to enable dimming for
+    # more models, e.g. add "is_k2_base": "LED" once its PWM support is confirmed.
+    LED_PIN_BY_MODEL: dict[str, str] = {
+        "is_k2_pro": "LED",
+        "is_k2_plus": "LED",
+    }
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)